libqrexec/exec.c

```diff
@@ -19,83 +19,118 @@
  *
  */
 
-#include <sys/socket.h>
-#include <sys/types.h>
 #include <stdlib.h>
-#include <unistd.h>
 #include <stdio.h>
 #include <string.h>
+#include <assert.h>
+#include <errno.h>
+#include <stddef.h>
+#include <limits.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <fcntl.h>
 #include "qrexec.h"
 #include "libqrexec-utils.h"
 
 static do_exec_t *exec_func = NULL;
 void register_exec_func(do_exec_t *func) {
+    if (exec_func == NULL)
```

Comment:
This is wrong: `exec_func` is the current value. It should be NULL before `register_exec_func()` set its it below.

Reply:
Fixed.